cmd/stats/stats.go

```diff
@@ -2,28 +2,223 @@ package stats
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"strings"
+
+	. "github.com/NicksPatties/sweet/db"
 	"github.com/spf13/cobra"
 )
 
-var Command = &cobra.Command{
+var Cmd = &cobra.Command{
 	Use:   "stats",
 	Short: "Print statistics about typing exercises",
 	Run: func(cmd *cobra.Command, args []string) {
-		since, _ := cmd.Flags().GetString("since")
-		printStats(since)
+		q, err := argsToQuery(cmd, time.Now())
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		reps, err := queryToReps(q)
+		if err != nil {
+			log.Fatal()
+			os.Exit(-1)
+		}
+		cols := argsToColumnFilter(cmd)
+		printStats(reps, cols)
 	},
 }
 
-func printStats(since string) {
-	if since != "" {
-		fmt.Printf("Printing stats since %s\n", since)
+// Returns the correct date depending on the start argument.
+// The time returned is at midnight of the date specified by `start`,
+// unless an hour query is used, in which case it will return the time
+// specified by
+func parseDateFromArg(isStart bool, arg string, now time.Time) (time.Time, error) {
+	var todayAtMidnight time.Time
+	todayAtMidnight = time.Date(now.Year(), now.Month(), now.Day(),
+		0, 0, 0, 0, now.Location())
+
+	if !isStart {
+		todayAtMidnight = todayAtMidnight.AddDate(0, 0, 1).Add(-1 * time.Nanosecond)
+	}
+
+	if arg == "" {
+		return todayAtMidnight, nil
+	}
+
+	unit := rune(arg[len(arg)-1])
+	amount, err := strconv.Atoi(arg[:len(arg)-1])
+
+	if err != nil {
+		// maybe it's in date format?
+		argTime, err := time.Parse(time.DateOnly, arg)
+
+		if err != nil {
+			return todayAtMidnight, fmt.Errorf("error parsing date: %s", err)
+		}
+
+		argTime = time.Date(
+			argTime.Year(), argTime.Month(), argTime.Day(),
+			0, 0, 0, 0, now.Location())
+
+		if !isStart {
```

Comment:
Either change `isStart` variable to `isEnd`, or assign a variable called `isEnd` at the beginning of the function